backend/apps/emails/models.py

```diff
@@ -0,0 +1,154 @@
+        _("Created at"),
+        auto_now=True,
+    )
+    changed_at = models.DateTimeField(
+        _("Last updated"),
+        auto_now_add=True,
+    )
+    subject = models.CharField(
+        _("Subject"),
+        max_length=255,
+        help_text=_("HTML content (may contain template variables, read body docs)."),
+        blank=True,
+        null=True,
+    )
+    body = models.TextField(
+        _("HTML template"),
+        help_text=_(
+            "HTML content (may contain template variables). You can use special "
+            "codes that will be substituted with appropriate information. This "
+            "template supports Candidates and Vacancies.<br/><br/>"
+            "<b>Candidates</b><br/>-----------<br/>"
+            "{{cfullname}} stands for candidate's full name<br/>"
+            "{{cname}} stands for candidate's firstname<br/>"
+            "{{csurname}} stands for candidate's surname<br/>"
+            "{{cage}} stands for candidate's age<br/><br/>"
+            "<b>Vacancy</b><br/>-----------<br/>"
+            "{{vtitle}} stands for vacancy title<br/>"
+            "{{vetype}} stands for vacancy employment type<br/>"
+            "{{vlocation}} stands for vacancy location<br/>"
+            "{{vel}} stands for vacancy level of english<br/>"
+            "{{vme}} stands for minimal experience for vacancy<br/>"
+            "{{vsd}} stands for vacancy start date<br/>"
+            "{{ved}} stands for vacancy end date<br/>"
+            "{{vdes}} stands for vacancy description<br/>"
+            "{{vsalmin}} stands for vacancy minimal salary<br/>"
+            "{{vsalmax}} stands for vacancy maximal salary<br/>"
+            "{{vsalcur}} stands for vacancy salary currency<br/>"
+            "-----------<br/><br/>You can use your own, which you can later add "
+            "while sending emails."
+        ),
+    )
+
+    def __str__(self) -> str:
+        """Return a readable name for the object."""
+        return f"{self.name} [{self.subject}]"
+
+    def __repr__(self) -> str:
+        """Return a detailed info on the object."""
+        return f"<EmailTemplate id={self.id} name='{self.name}'>"
+
+    def render_body(
+        self, context: dict, candidate: Candidate = None, vacancy: Vacancy = None
+    ) -> str:
+        """This method renders body for the message that can be
+        later used as HTML attachment for EmailMultiAlternatives.
+        """
+        return Template(self.body).render(
+            Context(context | self._form_keywords(candidate, vacancy))
+        )
+
+    def render_subject(
+        self, context: dict, candidate: Candidate = None, vacancy: Vacancy = None
+    ) -> str:
+        """Renders subjects for the email."""
+        return Template(self.subject).render(
+            Context(context | self._form_keywords(candidate, vacancy))
+        )
+
+    def _form_keywords(self, candidate: Candidate = None, vacancy: Vacancy = None) -> dict:
```

Comment:
Maybe rename the method to `_generate_keywords()`, because each time I see naming like "form_..." I think about django.forms method like 'form_valid()'